```{css, echo=FALSE}
body .main-container {
  max-width: 1280px !important;
  width: 1280px !important;
}
body {
  max-width: 1280px !important;
}
```
    
## Literature on Evolution of TRs and disorder



| Citation      |     Content |   Methods   |
| :-------- | :---------------------------------------------------------------------| :--------------------------------------------------------------------- |
| Simon, M., & Hancock, J. M. (2009). Tandem and cryptic amino acid repeats accumulate in disordered regions of proteins. Genome Biology.    |    |  Data: In-house homorepeat detector (l=1[aa]) on 5,815 orthologous proteins from four mammalian (human, chimpanzee, mouse and rat) and a bird (chicken) genome.  |
| Tompa, P. (2003). Intrinsically unstructured proteins evolve by repeat expansion. BioEssays : News and Reviews in Molecular, Cellular and Developmental Biology, 25(9), 847–855. http://doi.org/10.1002/bies.10324 | The frequency of TRs is higher in unstructured proteins compared to a background of all proteins. Tompa claims that repeat expansion is a major mechanism of the evolution of unstructured proteins. Tompa's view stems from the observatation of fast evolving micro- and minisatellites. | Unpublished algorithm also used by Marcotte & Pellegrini used for TR detection on 126 unstructured proteins and domains. |
| Jorda, J., Xue, B., Uversky, V. N., & Kajava, A. V. (2010). Protein tandem repeats - the more perfect, the less structured. FEBS Journal, 277(12), 2673–2682. http://doi.org/10.1111/j.1742-4658.2010.07684.x | "greater degrees of disorder in a protein sequence are associated with more perfect tandem repeats, or those with more precise spacing of the same amino acids across repeats" | TRs were predicted using T-REKS on PDB and SwissProt. Disorder was predicted using vlxt, vls2, iupred, foldindex, and TopIDP. |
|Szalkowski, A. M., & Anisimova, M. (2011). Markov models of amino acid substitution to study proteins with intrinsically disordered regions. PloS One, 6(5), e20488. http://doi.org/10.1371/journal.pone.0020488| | Data: T-REKS (de novo repeat detection algorithm, detects mostly TRs l < 15 [aa]) on Disprot |
| Lobanov, M. Y., & Galzitskaya, O. V. (2012). Occurrence of disordered patterns and homorepeats in eukaryotic and bacterial proteomes. Molecular BioSystems, 8(1), 327–337. http://doi.org/10.1039/c1mb05318c | Introducing the concept of "disordered patterns" - short sequences of amino acid, which are typically annotated as disordered in PDB. Calculating the frequency of these pattern over many proteomes. Correlating the frequency of homorepeats based on different amino acids across species -> There is a strong correlation for phylogenetically close species. | |
| Lobanov, M. Y., & Galzitskaya, O. V. (2015). How Common Is Disorder? Occurrence of Disordered Residues in Four Domains of Life. International Journal of Molecular Sciences, 16(8), 19490–19507. http://doi.org/10.3390/ijms160819490 | Calculating "Fraction of Disordered Residues in Proteins Containing Homo-Repeats of Different Length". Result: The fraction of disordered residues out of all residues in a protein depends on if the protein contain a homorepeat, and what kind of homorepeat. HOWEVER, the authors only look at this correlation, and not at the actual overlap, or any other confounding statistics. | 
| Kumari, B., Kumar, R., & Kumar, M. (2015). Low complexity and disordered regions of proteins have different structural and amino acid preferences. Molecular BioSystems, 11(2), 585–594. http://doi.org/10.1039/c4mb00425f | | 
| van der Lee, R., Buljan, M., Lang, B., Weatheritt, R. J., Daughdrill, G. W., Dunker, A. K., et al. (2014). Classification of Intrinsically Disordered Regions and Proteins. Chemical …, 114(13), 6589–6631. http://doi.org/10.1021/cr400525m | Large review on IDPs and IDRs. Concerning TRs, the prevalence of microsatellites in IDPs/IDRs is mentioned. The functional diversification following TR expansion is classified into three types. Tandem repeat expansion is seen as the origin of disordered region (with little explanation). | |
|Aït-Bara, S., Carpousis, A. J., & Quentin, Y. (2015). RNase E in the γ-Proteobacteria: conservation of intrinsically disordered noncatalytic region and molecular evolution of microdomains. Molecular Genetics and Genomics : MGG, 290(3), 847–862. http://doi.org/10.1007/s00438-014-0959-5 | | |
| Pellegrini, M. (2015). Tandem Repeats in Proteins: Prediction Algorithms and Biological Role. Frontiers in Bioengineering and Biotechnology, 3, 143. http://doi.org/10.3389/fbioe.2015.00143 | | |
| Roberts, S., Dzuricky, M., & Chilkoti, A. (2015). Elastin-like polypeptides as models of intrinsically disordered proteins. FEBS Letters, 589(19 Pt A), 2477–2486. http://doi.org/10.1016/j.febslet.2015.08.029 | Example of an artificial IDP governed by a pentapeptide TR. "Another common aspect that many IDPs and IDRs share with ELPs is the frequency of low sequence complexity tandem repeats, defined as patterns of repeats of highly similar amino acids [28,37]. The presence of low sequence complexity regions is unsurprising given the inherent bias toward disorder promoting amino acids. These regions in IDPs can be composed of simple repeats such as polyserine or polyglutamine or more complex repeats such as those found in resilin" | |
| Aït-Bara, S., Carpousis, A. J., & Quentin, Y. (2015). RNase E in the γ-Proteobacteria: conservation of intrinsically disordered noncatalytic region and molecular evolution of microdomains. Molecular Genetics and Genomics : MGG, 290(3), 847–862. http://doi.org/10.1007/s00438-014-0959-5 | | |
| Kajava, A. V., Klopffleisch, K., Chen, S., & Hofmann, K. (2014). Evolutionary link between metazoan RHIM motif and prion-forming domain of fungal heterokaryon incompatibility factor HET-s/HET-s. Scientific Reports, 4, 7436. http://doi.org/10.1038/srep07436 | | |
| Simon, A. J., Vallée-Bélisle, A., Ricci, F., & Plaxco, K. W. (2014). Intrinsic disorder as a generalizable strategy for the rational design of highly responsive, allosterically cooperative receptors. Proceedings of the National Academy of Sciences of the United States of America, 111(42), 15048–15053. http://doi.org/10.1073/pnas.1410796111 | | |
| Luo, H., & Nijveen, H. (2014). Understanding and identifying amino acid repeats. Briefings in Bioinformatics, 15(4), 582–591. http://doi.org/10.1093/bib/bbt003 | | |
| Espada, R., Parra, R. G., Sippl, M. J., Mora, T., Walczak, A. M., & Ferreiro, D. U. (2015). Repeat proteins challenge the concept of structural domains. Biochemical Society Transactions, 43(5), 844–849. http://doi.org/10.1042/BST20150083 | | |
| Mary, R. D., & Selvaraj, S. (2013). Analysis of sequence repeats of proteins in the PDB. Computational Biology and Chemistry, 47, 156–166. http://doi.org/10.1016/j.compbiolchem.2013.09.001 | | |
| Persi, E., & Horn, D. (2013). Systematic analysis of compositional order of proteins reveals new characteristics of biological functions and a universal correlate of macroevolution. PLoS Computational Biology, 9(11), e1003346. http://doi.org/10.1371/journal.pcbi.1003346 | | |
| Turjanski, P., Parra, R. G., Espada, R., Becher, V., & Ferreiro, D. U. (2015, October 8). Protein Repeats from First Principles. arXiv.org. | | |



#### [only]: Tandem repeats in general:

| Citation      |     Content |   Methods   |
| :-------- | :---------------------------------------------------------------------| :--------------------------------------------------------------------- |
| Marcotte, E. M., Pellegrini, M., Yeates, T. O., & Eisenberg, D. (1999). A census of protein repeats. Journal of Molecular Biology, 293(1), 151–160. http://doi.org/10.1006/jmbi.1999.3136 | First large review on protein tandem repeats | In house TR detector on Swissprot |
| Richard, G.-F., Kerrest, A., & Dujon, B. (2008). Comparative genomics and molecular dynamics of DNA repeats in eukaryotes. Microbiology and Molecular Biology Reviews : MMBR, 72(4), 686–727. http://doi.org/10.1128/MMBR.00011-08 | An excellent review on micro and minisatellites. No focus on protein TRs. Shortly discussing the distribution of short TRs in different species. "Interestingly, some of these regions also correspond to regions highly biased for GC content" | |
|Matsushima, N., Tanaka, T., & Kretsinger, R. (2009). Non-Globular Structures of Tandem Repeats in Proteins. Protein & Peptide Letters, 16(11), 1297–1322. doi:10.2174/092986609789353745 url to share this paper: sci-hub.tw/10.2174/092986609789353745 | Describes two classes of TRs: Globular and non-globular structures. Globulars can be divided into two subclasses. One which forms helix and solenoids ( ankyrin, HEAT, leucine rich repeat, and armadillo) and the other which has relatively stable domains (zinc-fingers, EF-hands, WD repeats, and SUSHI domain (SCR repeat)). The non-globular repeats are also found in two groups: those that form stable helices, like collagen, and those whose structures are flexible or inherently disordered, for example polyglutamine (polyGln)| | 


#### [only]: Tandem repeat structure:

| Citation      |     Content |   Methods   |
| :-------- | :---------------------------------------------------------------------| :--------------------------------------------------------------------- |
| Paladin, L., & Tosatto, S. C. E. (2015). Comparison of protein repeat classifications based on structure and sequence families. Biochemical Society Transactions, 43(5), 832–837. http://doi.org/10.1042/BST20150079 | Review on the structural classifications of TRs, very similar to Kajava's work. Finding significant overlap of PFAM annotations to structural classes in RepeatDB. | |


#### [only] IDPs:

| Citation      |     Content |   Methods   |
| :-------- | :---------------------------------------------------------------------| :--------------------------------------------------------------------- |
| Fuxreiter, M., & Tompa, P. (2012). Fuzzy Complexes: A More Stochastic View of Protein Function. In Fuzziness (pp. 1–14). Springer US. http://doi.org/10.1007/978-1-4614-0659-4_1 |  | |



#### Off-topic:

| Citation      |     Content |   Methods   |
| :-------- | :---------------------------------------------------------------------| :--------------------------------------------------------------------- |
| Rowling, P. J. E., Sivertsson, E. M., Perez-Riba, A., Main, E. R. G., & Itzhaki, L. S. (2015). Dissecting and reprogramming the folding and assembly of tandem-repeat proteins. Biochemical Society Transactions, 43(5), 881–888. http://doi.org/10.1042/BST20150099 | Review of known models for the order of folding of domain TRs (e.g. ANK, TPR, HEAT TRs). No mentioning of disordered regions, just misfolded, not-yet folded or unfolded TRs. | |
